modules/UI/UI.js

```diff
@@ -1106,6 +1154,62 @@ UI.showExtensionRequiredDialog = function (url) {
             "dialog.firefoxExtensionPrompt", {url: url}));
 };
 
+/**
+ * Shows dialog with information about camera or microphone error.
+ * @param {'microphone'|'camera'} type
+ * @param {JitsiTrackError} error
+ */
+UI.showDeviceErrorDialog = function (type, error) {
+    if (type !== "microphone" && type !== "camera") {
+        throw new Error("Invalid device type");
+    }
+
+    if (error.name && error instanceof JitsiMeetJS.JitsiTrackError &&
+        window.localStorage[type + "DoNotShowErrorAgain-" + error.name]
+            === "true") {
+        return;
+    }
+
+    let titleKey = error.name === TrackErrors.PERMISSION_DENIED
+            ? "dialog.permissionDenied"
+            : "dialog.error",
+        errorMsg = JITSI_TRACK_ERROR_TO_MESSAGE_KEY_MAP[type][error.name] ||
+            JITSI_TRACK_ERROR_TO_MESSAGE_KEY_MAP[type][TrackErrors.GENERAL],
+        title = `<span data-i18n="${titleKey}"></span>`,
+        message = "<h4 data-i18n='" + errorMsg + "'></h4>" +
+            (!JITSI_TRACK_ERROR_TO_MESSAGE_KEY_MAP[type][error.name]
+            && error.message
+                ? "<div>" + error.message + "</div>"
+                : "") +
+            "<label>" +
+                "<input type='checkbox' id='doNotShowWarningAgain'> " +
+                (error instanceof JitsiMeetJS.JitsiTrackError && error.name
+                    ? "<span data-i18n='dialog.doNotShowWarningAgain'></span>"
+                    : "") +
+            "</label>";
```

Comment:
The above is just hard to read because of:

1. The commas. Using separate lets for the separate variables would at least make it completely clear that one is dealing with multiple variables and not 1 variable with 1 very long rhs expression.

2. The inconsistent indentation. In this long "paragraph" there's a && at the beginning of the line. BUT in the preceding if the && was at the end of the line.

3. The repetition. JITSI_TRACK_ERROR_TO_MESSAGE_KEY_MAP[type] is there in this long "paragraph" 3 times. (Aside from it being unnecessary difficult to read, I hope the JS compiler is able to optimize these "3" property gets into 1.)